power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -427,6 +511,17 @@ template <bool sym> class YBus {
 
     // cache the math parameters
     std::shared_ptr<MathModelParam<sym> const> math_model_param_;
+
+    // cache the branch and shunt parameters in sequence_idx_map
+    IdxVector branch_param_idx_{};
+    IdxVector shunt_param_idx_{};
```

Comment:
is this caching still necessary?

Reply:
Yes, otherwise this get index of check will happen every single time

Reply:
Resolved in latest commit.